neo/test/coretest/test_spiketrain.py

```diff
@@ -313,6 +313,10 @@ def test__create_from_quantity_array(self):
 
     def test__create_from_quantity_array_with_dtype(self):
         times = np.arange(10, dtype="f4") * pq.ms
+        # this step is required for NumPy 2.0 which now casts to float64 in the case either value/array
+        # is float64 even if not necessary
+        # https://numpy.org/devdocs/numpy_2_0_migration_guide.html
+        times = times.astype(dtype="f4")
```

Comment:
Based on discussion with Sam my comment may not have been clear enough. The problem is that 
1* pq.ms is at baseline at `'float64'` and as of numpy 2.0+ everything is up-cast to the bigger dtype even if not necessarily. So in old versions of our test `times` would remain a `float32`, but with new versions of numpy make it into a `'float64'`. I did the astype to keep our original idea explict, but we could think of other rewrites of this test to maintain the dtype test.